### PR TITLE
[REF] mail, *: rename various JS models to match PY name

### PR DESCRIPTION
**Pull request by seb-odoo**
Move them to the correct folder (not full move, some references are still in the main folder and will have to be moved later).

And also fix related type hints.

Part of task-3605717

https://github.com/odoo/enterprise/pull/71221

### CODE REVIEW

File: addons/cloud_storage/controllers/attachment.py [hunk 1]
```diff
@@ -27,6 +27,6 @@ def mail_attachment_upload(self, ufile, thread_id, thread_model, is_pending=Fals
 
         # append upload url to the response to allow the client to directly
         # upload files to the cloud storage
-        attachment = request.env["ir.attachment"].browse(data["data"]["Attachment"][0]["id"]).sudo()
+        attachment = request.env["ir.attachment"].browse(data["data"]["ir.attachment"][0]["id"]).sudo()
```
Comment (seb-odoo) [inline addons/cloud_storage/controllers/attachment.py:30]: Note: this should probably be fixed in earlier version.

**Review by alexkuhn — APPROVED** [review]
LGTM

@robodoo delegate+ override=ci/security

`insert({ html: true })` on message and activity payload, which only affect html fields `message.body` and `activity.note`, both of which are sanitised

**Comment by alexkuhn** [comment]
@robodoo rebase-ff

**Comment by alexkuhn** [comment]
@robodoo r+

**Pull request closed without merging**